src/main/java/io/rpg/controller/Controller.java

```diff
@@ -58,10 +70,38 @@ public void setView(Scene currentView) {
     this.currentView = currentView;
   }
 
-  public void registerToViews(List<GameObjectView> views) {
-    for (GameObjectView view : views) {
-      view.addOnClickedObserver(this);
+  public void onAction(Action action) {
+    Class<?>[] args = {action.getClass()};
+    Method onSpecificAction;
+
+    try {
+      onSpecificAction = this.getClass().getDeclaredMethod("onAction", args);
+    } catch (NoSuchMethodException e) {
+      logger.error("onAction for " + action.getClass().getSimpleName() + "is not implemented");
+      return;
+    }
+
+    try {
+      onSpecificAction.invoke(this, action);
+    } catch (IllegalAccessException | InvocationTargetException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void onAction(LocationChangeAction action) {
+    if (!this.tagToLocationModelMap.containsKey(action.destinationLocationTag)) {
+      logger.error("Unknown location tag");
+      return;
     }
+
+    LocationView nextView = this.tagToLocationViewMap.get(action.destinationLocationTag);
+    LocationModel nextModel = this.tagToLocationModelMap.get(action.destinationLocationTag);
+
+    playerController.teleportTo(nextModel, nextView, action.playerPosition);
```

Comment:
we could also check here if the location is unlocked, for example whether the player has enough points or a specific item in his equipment